apps/notifications/src/app/templates/comment-reply-input.tsx

```diff
@@ -0,0 +1,261 @@
+						replyInputRef.current?.focus();
+						debug( 'Failed to submit comment reply: %s', error.message );
+					}
+					return;
+				}
+
+				if ( note.meta.ids.comment ) {
+					// pre-emptively approve the comment if it wasn't already
+					dispatch( actions.notes.approveNote( note.id, true ) );
+					restClient?.getNote( note.id );
+				}
+
+				// remove focus from textarea so we can resume using keyboard
+				// shortcuts without typing in the field
+				replyInputRef.current?.blur();
+
+				setValue( '' );
+				setIsSubmitting( false );
+				setHasClicked( false );
+				setRowCount( 1 );
+				retryCountRef.current = 0;
+
+				// remove the comment reply from local storage
+				repliesCache.removeItem( savedReplyKey );
+			} );
+		},
+		[ restClient, note, savedReplyKey, dispatch ]
+	);
+
+	const handleChange = ( value: string ) => {
+		const textarea = replyInputRef.current;
+		if ( ! textarea ) {
+			return;
+		}
+
+		dispatch( actions.ui.disableKeyboardShortcuts() );
+
+		setValue( value );
+		setRowCount( ( current ) =>
+			Math.min( 4, Math.ceil( ( textarea.scrollHeight * current ) / textarea?.clientHeight ) )
+		);
+
+		// persist the comment reply on local storage
+		if ( savedReplyKey ) {
+			repliesCache.setItem( savedReplyKey, [ value, Date.now() ] );
+		}
+	};
+
+	const handleClick = () => {
+		dispatch( actions.ui.disableKeyboardShortcuts() );
+
+		if ( ! hasClicked ) {
+			setHasClicked( true );
+		}
+	};
+
+	const handleFocus = () => {
+		dispatch( actions.ui.disableKeyboardShortcuts() );
+	};
+
+	const handleBlur = () => {
+		dispatch( actions.ui.enableKeyboardShortcuts() );
+
+		// Reset the field if there's no valid user input
+		// The regex strips whitespace
+		if ( '' === value.replace( /^\s+|\s+$/g, '' ) ) {
+			setValue( '' );
+			setHasClicked( false );
+			setRowCount( 1 );
+		}
+	};
+
+	const insertSuggestion = ( suggestion: { user_login: string }, suggestionsQuery: string ) => {
+		if ( ! suggestion ) {
+			return;
+		}
+
+		const element = replyInputRef.current;
+		const caretPosition = element?.selectionStart ?? 0;
+		const startString = value.slice( 0, Math.max( caretPosition - suggestionsQuery.length, 0 ) );
+		const endString = value.slice( caretPosition );
+
+		setValue( startString + suggestion.user_login + ' ' + endString );
+
+		element?.focus();
+
+		// move the caret after the inserted suggestion
+		const insertPosition = startString.length + suggestion.user_login.length + 1;
+		setTimeout( () => element?.setSelectionRange( insertPosition, insertPosition ), 0 );
+	};
+
+	useEffect( () => {
+		dispatch( actions.ui.disableKeyboardShortcuts() );
+		return () => {
+			dispatch( actions.ui.enableKeyboardShortcuts() );
+		};
+	}, [ dispatch ] );
+
+	useEffect( () => {
+		const handleKeyDown = ( event: KeyboardEvent ) => {
+			if ( ! keyboardShortcutsAreEnabled ) {
+				return;
+			}
+
+			if ( modifierKeyIsActive( event ) ) {
+				return;
+			}
+
+			if ( 82 === event.keyCode ) {
+				/* 'r' key */
```

Comment:
Linter complains `keyCode` is deprecated. Seems like an ok to clean it up.


```suggestion
			if ( 'r' === event.key.toLowerCase() ) {
```